### PR TITLE
DCOS-40526: add region information getter to pod struct

### PR DESCRIPTION
**Pull request by juliangieseke**
Closes DCOS-40526

## Testing

ui isnt implemented yet - `npm run test` it is.

## Trade-offs

I decided to just duplicate the code instead of extracting something into a util. I dont think that extracting it would add much value.

Also, this is a fixup commit for the last one, we're on a feature branch here which I'll have to rebase anyway - it _should_ work. I'm aware that this messes up our JIRAs a bit because there is no commit closing DCOS-40526 - lets see 🤷‍♂️

### CODE REVIEW

File: plugins/services/src/js/structs/Pod.js [hunk 1]
```diff
@@ -208,4 +208,28 @@ module.exports = class Pod extends Service {
...
+          const region = instance.getAgentRegion();
+          if (region !== "") {
+            regions[region] = regions[region] ? regions[region] + 1 : 1;
+          }
+
+          return regions;
+        }, {});
+
+      this._regions = Object.keys(regionCounts).sort(
+        (a, b) => regionCounts[b] - regionCounts[a]
```
Comment (nLight) [inline plugins/services/src/js/structs/Pod.js]: wonder why not natural sort? Couldn't find any explanation in the jira

Comment (juliangieseke) [reply plugins/services/src/js/structs/Pod.js]: yeah, was my decision - I thought its a good idea to sort by tasks running on that instance 🤷‍♂️

Comment (nLight) [reply plugins/services/src/js/structs/Pod.js]: Couldn't it be confusing for a user as we don't show the actual count?

Comment (juliangieseke) [reply plugins/services/src/js/structs/Pod.js]: The actual column is added here: https://github.com/dcos/dcos-ui/pull/3167
I'll change the implementation to use natural sort - have to rebase the feature branch anyway 🤷‍♂️

**Pull request merged**